internal/kmsg/kmsg.go

```diff
@@ -0,0 +1,63 @@
+// Package kmsg contains support for parsing Linux kernel log entries read from
+// /dev/kmsg. These are the same log entries that can be read via the `dmesg`
+// command. Each read from /dev/kmsg is guaranteed to return a single log entry,
+// so no line-splitting is required. If the read buffer is not large enough to
+// contain the log entry, `os.ErrInvalid` is returned.
+//
+// More information can be found here:
+// https://www.kernel.org/doc/Documentation/ABI/testing/dev-kmsg
+package kmsg
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+)
+
+// Entry is a single log entry in kmsg.
+type Entry struct {
+	SyslogPriority     uint8
+	SyslogFacility     uint8
+	Seq                uint64
+	TimeSinceBootMicro uint64
+	Flags              string
+	Message            string
+}
+
+var (
+	// ErrInvalidFormat indicates the kmsg entry failed to parse.
+	ErrInvalidFormat = errors.New("invalid kmsg format")
+)
+
+// Parse takes a single kmsg log entry string and returns a struct representing
+// the components of the log entry.
+func Parse(s string) (*Entry, error) {
+	fields := strings.SplitN(s, ";", 2)
+	if len(fields) < 2 {
+		return nil, ErrInvalidFormat
+	}
+	prefixFields := strings.SplitN(fields[0], ",", 5)
```

Comment:
can you explain what you're doing here/how a line is expected to be formatted like?

Reply:
Update: this is explained in the above comment which links to [this](https://www.kernel.org/doc/Documentation/ABI/testing/dev-kmsg}